src/nixpkgs_tree_docsource.rs

```diff
@@ -64,7 +50,14 @@ impl DocSource for NixpkgsTreeDatabase {
             .map(|k| DocEntry::NixpkgsTreeDoc(k.clone()))
             .collect()
     }
+    fn update(&mut self) -> Result<bool, Errors> {
+        let new_keys = gen_keys()?;
+        let last = std::mem::replace(&mut self.keys, new_keys);
+
+        Ok(last != self.keys)
```

Comment:
Generally i see the point in having `update` return information about if anything actually changed, but in cases like this, doing that check might actually have some performance implications (as it probably needs to iterate over all of the values pretty much, "undoing" the efficiency you've saved by doing the `std::mem::replace`. Are you sure that we actually _need_ that return information, or that this causes no noticable increase in update-times?